packages/compass-aggregations/src/stores/store.ts

```diff
@@ -0,0 +1,275 @@
+   * `fields-changed` events and updates corresponding redux slices
+   * accordingly
+   */
+  localAppRegistry: Pick<AppRegistry, 'on' | 'emit' | 'getStore'>;
+  /**
+   * Instance of global app registry, listens to `refresh-data` and
+   * `import-finished` events and updates corresponding redux slices
+   * accordingly
+   */
+  globalAppRegistry: Pick<AppRegistry, 'on' | 'emit' | 'getStore'>;
+  /**
+   * Should be provided if namespace is a view. Affects available stages
+   */
+  sourceName: string;
+  /**
+   * Current server version. Affects available stages
+   */
+  serverVersion: string;
+  /**
+   * Whether or not collection is a timeseries collection. Affects available
+   * stages
+   */
+  isTimeSeries: boolean;
+  /**
+   * Whether or not collection is part of ADF (ex ADL). Used only to provide
+   * correct explain plan verbosity
+   */
+  isDataLake: boolean;
+  /**
+   * Current connection env type. Affects available stages. Accepted values:
+   * "atlas" | "on-prem" | "adl"
+   */
+  env: typeof ENVS[number] | null;
+  /**
+   * Indicates that the plugin is used in Atlas Cloud
+   */
+  isAtlasDeployed: boolean | null;
+  /**
+   * Namespace field values that will be used in autocomplete
+   */
+  fields: { name: string }[];
+  /**
+   * Function that overrides default handling of opening resulting namespace
+   * of out stages ($out, $merge). When provided, will be called when user
+   * clicks
+   */
+  outResultsFn: (namespace: string) => void;
+  /**
+   * Stored pipeline metadata. Can be provided to preload stored pipeline
+   * right when the plugin is initialized
+   */
+  aggregation: StoredPipeline;
+  /**
+   * Namespace for the view that is being edited. Needs to be provided
+   * with the `sourcePipeline` options. Takes precedence over `pipeline`
+   * option
+   */
+  editViewName: string;
+  /**
+   * Pipeline definition for the view that is being edited. Needs to be
+   * provided with the `editViewName` option. Takes precedence over
+   * `pipeline` option
+   */
+  sourcePipeline: unknown[];
+  /**
+   * Initial pipeline that will be converted to a string to be used by the
+   * aggregation builder. Takes precedence over `pipelineText` option
+   */
+  pipeline: unknown[];
+  /**
+   * Initial pipeline text to be used by the aggregation builder
+   */
+  pipelineText: string;
+}>;
+
+const configureStore = (options: ConfigureStoreOptions) => {
+  if (!options.dataProvider?.dataProvider) {
+    throw new Error(
+      "Can't configure store for aggregation plugin without data serivce"
+    );
+  }
+
+  if (options.editViewName && !options.sourcePipeline) {
+    throw new Error(
+      'Option `editViewName` can be used only if `sourcePipeline` is provided'
+    );
+  }
+
+  const editingView = !!(options.editViewName && options.sourcePipeline);
+
+  const initialPipelineSource =
+    (editingView
+      ? toJSString(options.sourcePipeline, '  ')
+      : options.pipeline
```

Comment:
Ensuring i got it right, we added `pipeline` and `pipelineText` as options for public implementation, right?

Reply:
Yes, we don't really need it for any logic in the app (it's handy in tests though), just two options for external users to pass initial pipeline value instead of using `sourcePipeline` that has a bit of a different meaning for us